### PR TITLE
[#169872495] Review and improve cdn-route service custom domain docs

### PR DESCRIPTION
**Pull request by jonathanglassman**
What
----

Clarified content in https://docs.cloud.service.gov.uk/deploying_services/use_a_custom_domain/#managing-custom-domains-using-the-cdn-route-service to:

- provide an overview and context of the process
- better explain where a user might have to carry out the various tasks described (ie in DNS provider interface or the PaaS command line)
- link to GOV.UK guidance

How to review
-------------

Describe the steps required to test the changes.

1. Preview the content locally ([see README](https://github.com/alphagov/paas-tech-docs#preview)) and check that it renders as expected.
1. Manually check that any removed or renamed anchor tags are not in use ([linkchecker](https://github.com/linkcheck/linkchecker) doesn't do this).
1. Check the new content fulfils https://www.pivotaltracker.com/story/show/169872495 

Who can review
--------------

Anyone except Jon Glassman and Richard Towers

### CODE REVIEW

File: source/documentation/deploying_services/use_a_custom_domain.erb [hunk 1]
```diff
@@ -1,32 +1,43 @@
...
+- use your organisation's preferred domain name system (DNS) provider to host DNS
+- manage the domain's TLS certificates
+- route requests to GOV.UK PaaS
+- cache responses (optional)
 
-You must configure the cdn-route service to use a subdomain. If you configure the service to use an apex domain, the service creation might not succeed. The following table contains examples of apex domains and subdomains:
+GOV.UK PaaS does not provide domain registration or DNS hosting. Use your organisation's preferred registrar and DNS provider.
 
-<div style="height:1px;font-size:1px;">&nbsp;</div>
+You can use either the GOV.UK PaaS [cdn-route service](https://github.com/alphagov/paas-cdn-broker) or use a commercial content distribution network (CDN) to manage TLS and route requests to GOV.UK PaaS.
```
Comment (46bit) [inline source/documentation/deploying_services/use_a_custom_domain.erb]: General feedback and not something we need to fix: It's pleasantly transparent to link to our https://github.com/alphagov/paas-cdn-broker repo, but I don't think its README is very tenant-friendly.

File: source/documentation/deploying_services/use_a_custom_domain.erb [hunk 2]
```diff
@@ -252,7 +259,7 @@ cf delete-service SERVICE_INSTANCE
 
 where `SERVICE_INSTANCE` is the name of the appropriate cdn-route service instance.
 
-You have now deleted your custom domain from the GOV.UK PAaS. You should next [change or remove the domain’s DNS records](#changing-your-custom-domain-39-s-dns-records).
+You have now deleted your custom domain from GOV.UK PAaS. You should next [change or remove the domain’s DNS records](#changing-your-custom-domain-39-s-dns-records).
```
Comment (46bit) [inline source/documentation/deploying_services/use_a_custom_domain.erb]:
```suggestion
You have now deleted your custom domain from GOV.UK PaaS. You should next [change or remove the domain’s DNS records](#changing-your-custom-domain-39-s-dns-records).
```

File: source/documentation/deploying_services/use_a_custom_domain.erb [hunk 3]
```diff
@@ -273,12 +280,12 @@ cf update-service custom-domains-production \
...
 
-- delete or amend the subdomain's CNAME so that it does not point to the GOV.UK PaaS
-- delete the subdomain's TXT record you created when you configured the subdomain to work with the GOV.UK PaaS
+- delete or amend the subdomain's CNAME so that it does not point to GOV.UK PaaS
+- delete the subdomain's TXT record you created when you configured the subdomain to work with GOV.UK PaaS
 
-<%= warning_text('Do not stop your domains` DNS records from pointing towards the GOV.UK PaaS before you have removed those custom domain(s) from your cdn-route service. <br><br> If you do, all of your domains` TLS certificates will fail to renew without warning at some point over the next 90 days. Your users may not be able to use your service.') %>
+<%= warning_text('Do not stop your domains` DNS records from pointing towards GOV.UK PaaS before you have removed those custom domain(s) from your cdn-route service. <br><br> If you do, all of your domains` TLS certificates will fail to renew without warning at some point over the next 90 days. Your users may not be able to use your service.') %>
```
Comment (46bit) [inline source/documentation/deploying_services/use_a_custom_domain.erb]: This is best practice but it doesn't clearly explain what is going on.
```suggestion
<%= warning_text('Do not stop your domains` DNS records from pointing towards GOV.UK PaaS before you have removed those custom domain(s) from your cdn-route service. <br><br> If you do, all of the domains` associated with that cdn-route service with fail to renew their TLS certificates without warning at some point over the next 90 days. Your users may not be able to use your service.') %>
```

Comment (46bit) [reply source/documentation/deploying_services/use_a_custom_domain.erb]: Making clearer what the safe thing to do is:
```suggestion
<%= warning_text('Remove a custom domain from your cdn-route service before removing the DNS records pointing at GOV.UK PaaS. <br><br>Otherwise all of the domains` associated with that cdn-route service will fail to renew their TLS certificates without warning at some point over the next 90 days. Your users may not be able to use your service.') %>
```

**Comment by 46bit** [comment]
I'm OK with this. I think an extra redraft would be lovely but it snowballed when I tried it. I've sent off for a tech writer review ("pre-i.")

**Comment by 46bit** [comment]
Ready for review (and shipping.) We'll want to revisit this after Tech Docs training.

File: source/documentation/deploying_services/use_a_custom_domain.erb [hunk 4]
```diff
@@ -1,32 +1,30 @@
 # Managing custom domains using the cdn-route service
 
-This section explains how to manage custom domains for your app with a content distribution network (CDN). The CDN:
+Apps on GOV.UK PaaS are given a domain of `APP_NAME.london.cloudapps.digital` by default. This is OK to use within your team and for development. However, you shouldn’t use it for a live, public service. This guide will help you to use a custom domain instead.
```
Comment (AP-Hunt) [inline source/documentation/deploying_services/use_a_custom_domain.erb]: I'm not very sure about the "you shouldn't use it for". It has the wrong tone, for my money. I think we should rephrase it to talk about how a live service will need to be accessed. Something like

> However, a live service will need likely need its own domain name, such as `my-service.service.gov.uk`

File: source/documentation/deploying_services/use_a_custom_domain.erb [hunk 5]
```diff
@@ -36,12 +34,14 @@ Once you create a CDN service instance, you cannot update or delete the instance
...
 
-1. Create the required domain(s) in your organisation. Run the following for each domain:
+1. Create the required domain(s) in your organisation:
 
     ```
-    cf create-domain ORG_NAME DOMAIN_NAME
+    cf create-domain ORG_NAME APEX_DOMAIN_NAME
     ```
 
+    where `APEX_DOMAIN_NAME` is the name of the apex domain, for example `example.com`
```
Comment (AP-Hunt) [inline source/documentation/deploying_services/use_a_custom_domain.erb]: Could we change this to something like `example.service.gov.uk` to make it clear the apex can be a subdomain of an existing domain?

**Comment by AP-Hunt** [comment]
I've been asked by >1 tenant how they should go about redirecting `www.my.service` to `my.service` and whether the CDN route service can do it. Should we document how to do it properly here? (i.e. deploy an nginx buildpack app)

**Comment by 46bit** [comment]
> I've been asked by >1 tenant how they should go about redirecting www.my.service to my.service and whether the CDN route service can do it. Should we document how to do it properly here? (i.e. deploy an nginx buildpack app)

I agree we should but I simply refuse to expand the scope of this particular story any further… we've already done at least two rewrites. 😱

File: source/documentation/deploying_services/use_a_custom_domain.erb (hunk 4 above)
Comment (46bit) [reply source/documentation/deploying_services/use_a_custom_domain.erb]: Thanks Andy, I agree. I've pushed a commit which does this but I found it quite awkward to integrate. Re-review please?

[the remark this answers is not in the document]
Comment (46bit) [reply source/documentation/deploying_services/use_a_custom_domain.erb]: It isn't. Ireland uses `APP_NAME.cloudapps.digital`. Only London uses `APP_NAME.london.cloudapps.digital`. Trying to clean up that ambiguity made these docs more unreadable 😞 I'm going to ask about this at Tech Docs training next week.

**Comment by 46bit** [comment]
I think we should publish this content and redraft when it's done. @schmie's poked some definite holes in this content, but after one month I think we should declare 'good enough' and redraft it after we're Tech Docs trained?
